src/storage/gcloud-storage/storage.ts

```diff
@@ -0,0 +1,106 @@
+import type { Bucket } from '@google-cloud/storage';
+import type { Account } from 'account/account.entity';
+import type { AccountService } from 'account/account.service';
+import { type Result, error, getError, isError, ok } from 'core/result';
+import type { Context } from 'hono';
+import { v4 as uuidv4 } from 'uuid';
+
+const ALLOWED_IMAGE_TYPES = [
+    'image/jpg',
+    'image/jpeg',
+    'image/png',
+    'image/webp',
+];
+
+export type FileValidationError = 'file-too-large' | 'file-type-not-supported';
+
+export function createStorageHandler(
+    accountService: AccountService,
+    bucket: Bucket,
+) {
+    /**
+     * Handle an upload to GCloud Storage bucket
+     */
+    return async function handleUpload(ctx: Context) {
```

Comment:
This is a HTTP handler right? I think this should live in the `src/http` directory, and then use the storage service as a dependency

Reply:
Done in https://github.com/TryGhost/ActivityPub/pull/533/commits/06a1d0a69cccc4338b88638867afdd7091222c8f